restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/ua/UserAgentManager.java

```diff
@@ -662,17 +716,30 @@ private String createAddress(SipServletRequest request) throws ServletParseExcep
         } else {
             buffer.append("sip:");
         }
-        buffer.append(normalize(user)).append("@").append(uri.getHost()).append(":").append(uri.getPort());
+        int contactPort = contactUri.getPort();
+        if(contactPort < 0) {
+            if(transport == null || transport.equalsIgnoreCase("udp") || transport.equalsIgnoreCase("tcp")) {
+                contactPort = 5060;
```

Comment:
where are these ports coming from, shouldnt be coming from env configuration?

Reply:
this is the default SIP ports for the given transports if not present in the SIP URI. We may want to have a util class for that as I think there is a couple of other places in RC or SIP Servlets that does this.